pkg/controller/containersource/reconcile.go

```diff
@@ -106,9 +111,11 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) (runt
 			r.recorder.Eventf(source, corev1.EventTypeNormal, "Deployed", "Created deployment %q", deploy.Name)
 			source.Status.MarkDeploying("Deploying", "Created deployment %s", args.Name)
```

Comment:
Optional suggestion: what do you think about adding a `return source, nil` here, since this is effectively a terminal state for this reconcile pass? That allows the following `else` block to be elevated up one level, which seems closer to idiomatic go style.